src/AdaptAuth.ts

```diff
@@ -241,7 +240,18 @@ export class AdaptAuth {
   /**
    * Helper to extract the saml relay final destination url from req object
    */
-  public getFinalDestination = (req: any) => ((req as SamlUserRequest).samlRelayState || {}).finalDestination;
+  public getFinalDestination = (req: any) => {
+    // Attach relayState to req
+    try {
+      const relayState = JSON.parse(req.body.RelayState);
+      const finalDest = relayState.finalDestination || null;
+      return finalDest;
+
+    } catch (err) {
+      // I guess the relayState wasn't that great...
+      console.log('Unable to parse samlRelayState', err);
+    }
+  };
```

Comment:
Since we re-attached `samlRelayState` to the request object we could just get from there now rather than re-parsing

Reply:
I just ran a test replacing line 246 with :
```
const relayState = req.relayState;
```
and req.relayState is coming back as undefined and throwing an error.

Which refreshes my memory on why I removed the samlRelayState code on lines 65-70 and moved it down here to begin with. For whatever reason it wasn't on the request object when checked in getFinalDestination(). 

Since the getFinalDestination() seemed to be the only place the relayState was getting used, just parsing it here directly seemed like an appropriate solution.

Reply:
it should be [attached as](https://github.com/SU-SWS/adapt-auth-sdk/blob/ADAPT-4167/src/AdaptAuth.ts#L66)
```typescript
const relayState = req.samlRelayState;
```
While we may only be using it within this middleware for getting the final destination there is usefulness in pre-parsing the SAML relay state and providing it on the request on object to middleware further down the chain.

Reply:
Refactoring it to the correct name fixed this issue. Facepalm. 
Just pushed out a change for that.